internal/config/manager.go

```diff
@@ -0,0 +1,274 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"go.uber.org/zap"
+)
+
+const (
+	TombValue        = iota
+	CustomSourceName = "CustomSource"
+)
+
+type Manager struct {
+	sync.RWMutex
+	sources        map[string]Source
+	keySourceMap   sync.Map
+	overlayConfigs sync.Map // store the configs setted or deleted by user
+}
+
+func NewManager() *Manager {
+	return &Manager{
+		sources: make(map[string]Source),
+	}
+}
+
+func (m *Manager) GetConfig(key string) (string, error) {
+	realKey := formatKey(key)
+	v, _ := m.overlayConfigs.Load(realKey)
+	if v != nil {
+		if v == TombValue {
+			return "", fmt.Errorf("key not found: %s", key)
+		}
+		return v.(string), nil
+	}
```

Comment:
if user set a config override the os config, then user  delete it, GetConfig will not return the os config value ?

Reply:
yeap, the `DeleteConfig` means I do not want to see it anymore